File: Code Optimization and Debugging II/C++/CheckPalindromeNumber.cpp
#include <iostream>
using namespace std;

bool isPalindrome(int num) {
    int rev = 0, orig = num;

    while (num != 0) {
        rev = (rev * 10) + (num % 10);
        num /= 10;
    }

    return orig == rev;
}

int main() {
    int num;
    cin >> num;
    
    isPalindrome(num) ? printf("%d is a Palindrome", num) : printf("%d is Not Palindrome", num); 
}

/*

Output:

100001
100001 is a Palindrome

*/

File: Code Optimization and Debugging II/Python/CheckPalindromeNumber.py
def isPalindrome(num):
    rev = 0
    orig = num

    while num != 0:
        rev = (rev * 10) + (num % 10)
        num //= 10

    return rev == orig

num = int(input())
print("{} is a palindrome".format(num) if isPalindrome(num) else "{} is not a palindrome".format(num))
